documentation/adr/ADR-0004.md

```diff
@@ -0,0 +1,39 @@
+## Title
+
+Converting the repository into an apt-based application (.deb package)
+
+## Proposal status
+
+What is the status of this decision?
+
+-   Proposed
+
+## Implementation status
+
+-   Work in progress
+
+## Context
+
+Our current application utilizes CURL installation methood and the legacy installation method that is cloning the repository and installing each dependency individually. This requires frequent maintenance by the maintainer and necessitates the manual installation of the dependencies. Switching to packaging, specifically to ".deb" packages, would streamline this process by pre-processing all the dependencies and install the application with all the necessary package dependency. This will eliminate the need for manual intervention by the maintainer unless a new dependency was added/contributed on the open-source repository for each system.
+
+## Decision
+
+I propose converting the current CURL installation method to an APT-based installation. APT based installations are more secure as they involve signed packages, while CURL installations directly download from potentially unknown sources. Additionally, APT allows for easy rollback to previous versions in case a newer version is found to be buggy, providing a reliable way to manage software updates and maintain system stability. Moreover, APT-based installations are system-wide, making the software available to all users on the system.
+
+## Consequences
+
+### What will become easier:
+
+Dependency Management: APT will automatically handle and resolve dependencies, reducing the manual effort required during installation.
+
+Security: Packages installed via APT are signed and verified, enhancing the overall security of the installation process.
+
+Maintenance: Centralized package management through APT reduces the maintenance overhead for the development and support teams.
+
+### What will become difficult:
```

Comment:
There are several additional challenges not addressed here:

- Where will the APT repository be hosted?
- Who will maintain this repository now?
- Who will maintain it long term?
- For those not using `apt` as a package manager or a debian based OS, how will they use PT-GUI? If using the existing method, does creating a new installation medium not create additional overhead? Who will manage version control to ensure each medium is updated equally?